node/src/components/storage.rs

```diff
@@ -653,7 +653,8 @@ impl Storage {
                 let finality_signatures =
                     match self.get_finality_signatures(&mut txn, &block_hash)? {
                         Some(signatures) => signatures,
-                        None => BlockSignatures::new(block_hash, block.header().era_id()),
```

Comment:
@darthsiroftardis @marc-casperlabs If there are no block signatures, this should not return success...

Reply:
This needs further review.  This logic is used by the RPC, where this is not the correct behavior. However, a _fetcher_ should refuse to return a payload which does not finality signatures.